transactions/0062.md

```diff
@@ -179,62 +149,48 @@ console.log({ correctOrder: khanTopologicalSort(graph) })
-47304402206b1255cb23454c63b22833de25a3a3ecbdb8d8645ad129d3269cdddf10b2ec98022034cadf46e5bfecc38940e5497ddf5fa9aeb37ff5ec3fe8e21b19cbb64a45ec324121029a82bfce319faccc34095c8405896e1223921917501a4f736a04f126d6a01c12 // unlocking script
-ffffffff // nSequence
-// CONDITIONAL EXTENDED DATA
-    00 // indicates that no extended data follows
-// END OF INPUT
-01 // nOutputs
-0100000000000000 // satoshis
-19 // locking script length
-76a914d866ec5ebb0f4e3840351ee61887101e5407562988ac // locking script
-00000000 //  nLocktime
+// rawtx parent follows
+020000000158cb8b052fded9a6c450c4212562df8820359ec34da41286421e0d0f2b7eefee000000006a47304402206b1255cb23454c63b22833de25a3a3ecbdb8d8645ad129d3269cdddf10b2ec98022034cadf46e5bfecc38940e5497ddf5fa9aeb37ff5ec3fe8e21b19cbb64a45ec324121029a82bfce319faccc34095c8405896e1223921917501a4f736a04f126d6a01c12ffffffff0101000000000000001976a914d866ec5ebb0f4e3840351ee61887101e5407562988ac00000000
+01 // above tx has merkle path
+00 // VarInt the index of the path for this tx in the above list
+// rawtx current payment follows
+020000000158cb8b052fded9a6c450c4212562df8820359ec34da41286421e0d0f2b7eefee000000006a47304402206b1255cb23454c63b22833de25a3a3ecbdb8d8645ad129d3269cdddf10b2ec98022034cadf46e5bfecc38940e5497ddf5fa9aeb37ff5ec3fe8e21b19cbb64a45ec324121029a82bfce319faccc34095c8405896e1223921917501a4f736a04f126d6a01c12ffffffff0101000000000000001976a914d866ec5ebb0f4e3840351ee61887101e5407562988ac00000000
+00 // above tx doesn't have merkle path, but instead has local parent
 ```
 
-## Backward compatibility
-
-The Background Evaluation Extended Format is not backwards compatible, but has been designed in such a way that existing software should not read a transaction in Background Evaluation Extended Format as a valid (partial) transaction. The Background Evaluation Extended Format header (00000000BEEF) will be read as an empty transaction with a future nLock time in a library that does not support the Background Evaluation Extended Format. 
+## Validation Process
+
+1. We parse the Compound Merkle Paths (CMP) storing each in an array so that we can address them by index later.
+2. We parse each transaction.
+    a. RawTx bytes double sha256 to get the txid.
+    b. Store in hashmap from txid => parsedTx
+    c. If there is a Merkle path after the tx then we lookup the CMP using the CMP index number.
+        1. Lookup the txid within level 0 leaves of the CMP to get the index of the txid within a block.
+        2. Calculate the Merkle root with the index, txid, and CMP data.
+        3. Add the merkle root to an array which will be used in a request to our local header service once all transactions have been parsed.
+        4. Mark the tx as valid or not as soon as the header service responds.
+    d. Otherwise we run local validation on the transaction, stopping at the soonest failure.
+        1. Check the txid exists in memory, and is marked as valid.
+        2. Check that all scripts evaluate to TRUE.
```

Comment:
So the part of SPV will be running a unlocking/locking scripts?

Reply:
Yes